<a href="https://colab.research.google.com/github/nanoCHI/Data-Analysis/blob/main/Call_Volume_Trend_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Importing all the important libraries

In [4]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt


# Read CSV file

In [7]:
df = pd.read_csv('/content/drive/My Drive/Colab Notebooks/Call_Volume_Trend_Analysis_Project_9.xlsx - Call_Data.csv')
df

Agent_Name   Agent_ID Customer_Phone_No  Queue_Time(Secs)  \
0       Executives 42  1000042.0        98502XXXXX                 2   
1        Executives 4  1000004.0        80595XXXXX                 0   
2       Executives 65  1000065.0        70202XXXXX                 0   
3       Executives 55  1000055.0        96104XXXXX                 1   
4       Executives 21  1000021.0        82001XXXXX                 0   
...               ...        ...               ...               ...   
117983   Executives 7  1000007.0        85399XXXXX                62   
117984  Executives 36  1000036.0        88397XXXXX                22   
117985  Executives 48  1000048.0        70035XXXXX                14   
117986  Executives 33  1000033.0        87773XXXXX                26   
117987  Executives 58  1000058.0        83800XXXXX                 5   

       Date_&_Time  Time Time_Bucket Duration(hh:mm:ss)  Call_Seconds (s)  \
0         1/1/2022   9.0        9_10            0:01:36              96.0   
1         1/1/2022   9.0        9_10            0:02:20             140.0   
2         1/1/2022   9.0        9_10            0:01:25              85.0   
3         1/1/2022   9.0        9_10            0:01:31              91.0   
4         1/1/2022   9.0        9_10            0:02:45             165.0   
...            ...   ...         ...                ...               ...   
117983   1/23/2022  20.0       20_21            0:01:32              92.0   
117984   1/23/2022  20.0       20_21            0:01:14              74.0   
117985   1/23/2022  20.0       20_21            0:01:48             108.0   
117986   1/23/2022  20.0       20_21            0:01:13              73.0   
117987   1/23/2022  20.0       20_21            0:01:22              82.0   

       Call_Status  Wrapped _By Ringing IVR _Duration  
0         answered        Agent     YES       0:00:16  
1         answered        Agent     YES       0:00:26  
2         answered  AutoWrapped     YES       0:00:16  
3         answered        Agent     YES       0:00:25  
4         answered        Agent     YES       0:00:23  
...            ...          ...     ...           ...  
117983    answered        Agent     YES       0:00:42  
117984    answered        Agent     YES       0:00:15  
117985    answered        Agent     YES       0:00:23  
117986    answered        Agent     YES       0:01:05  
117987    answered        Agent     YES       0:00:14  

[117988 rows x 13 columns]

# Cleaning The DataFrame
1. Checking For Duplicate Values
2. Checking for irrelevant columns
3. Checking for null values

In [13]:
df.duplicated().sum()

941

> So there are 941 Duplicate Values
>> Next step is to remove those rows from the dataframe

In [17]:
df = df.drop_duplicates()

# Re-checking
if(df.duplicated().sum()==0):
  print("No more duplicate values")
else:
  print("More duplicates values found")

No more duplicate values
